Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Dialogs/Attachments/AttachmentDialog.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Dialogs.Choices;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.BotFrameworkFunctionalTests.TeamsWaterfallSkillBot.Dialogs.Attachments
+{
+    public class AttachmentDialog : ComponentDialog
+    {
+        private const string Picture = "architecture-resize.png";
+
+        public AttachmentDialog()
+             : base(nameof(AttachmentDialog))
+        {
+            AddDialog(new ChoicePrompt("AttachmentTypePrompt"));
+            AddDialog(new ChoicePrompt("EndPrompt"));
+            AddDialog(new WaterfallDialog(nameof(WaterfallDialog), new WaterfallStep[] { SelectAttachmentAsync, HandleAttachmentAsync, FinalStepAsync }));
+            InitialDialogId = nameof(WaterfallDialog);
+        }
+
+        private static Attachment GetInlineAttachment()
+        {
+            var imagePath = Path.Combine(Environment.CurrentDirectory, @"Dialogs/Attachments/Files", Picture);
+            var imageData = Convert.ToBase64String(File.ReadAllBytes(imagePath));
+
+            return new Attachment
+            {
+                Name = $@"Files\{Picture}",
+                ContentType = "image/png",
+                ContentUrl = $"data:image/png;base64,{imageData}",
+            };
+        }
+
+        private static Attachment GetInternetAttachment()
+        {
+            // ContentUrl must be HTTPS.
+            return new Attachment
+            {
+                Name = $@"Files\{Picture}",
+                ContentType = "image/png",
+                ContentUrl = "https://docs.microsoft.com/en-us/bot-framework/media/how-it-works/architecture-resize.png",
```

Comment:
Let's avoid external dependencies, move the image to the project so the tests are self contained.

Reply:
refactored